Fix batch rename for unique name and empty name

ajreckof opened the pull request:
Fixes #78290 
Fixes #78291

ajreckof commented on editor/gui/scene_tree_editor.h:
```diff
@@ -62,6 +62,7 @@ class SceneTreeEditor : public Control {
 	String filter_term_warning;
 
 	AcceptDialog *error = nullptr;
+	AcceptDialog *error2 = nullptr;
```
Both can appear at the same time if the name is invalid it is replacing the invalids characters with `_`. This means if you try to rename a node that is Unique to `Node%` and you already have a unique name called `Node_` it will first tells you there was invalid characters and then tells you this unique name is already used. Right now this specific case make the unique name be revoked even with normal renames because I think it checked them in the wrong order.

I'm not sure about adding a return to the first error as this is not how it behaved previously.

KoBeWi replied:
read my edit

ajreckof commented:
Implemented what kobewi suggested with both it now look like this.
<img width="454" alt="Capture d’écran 2023-06-20 à 23 22 22" src="https://github.com/godotengine/godot/assets/66184050/78e278b9-12e8-46b8-a8e5-7e78c48dd769">

KoBeWi commented on editor/gui/scene_tree_editor.cpp:
```diff
@@ -968,15 +968,34 @@ void SceneTreeEditor::_rename_node(Node *p_node, const String &p_name) {
-		error->set_text(TTR("Invalid node name, the following characters are not allowed:") + "\n" + String::get_invalid_node_name_characters());
-		error->popup_centered();
+		String text = TTR("Invalid node name, the following characters are not allowed:") + "\n" + String::get_invalid_node_name_characters();
+		if (error->is_visible()) {
```
Not sure if this condition is necessary; it's not really something that can happen here.

ajreckof replied:
Unfortunately this can happen with batch rename if one node is erroring because of unique name already in use and another node because of invalid character later.

KoBeWi replied:
> I'll try to show an actual case of it happening tomorrow.

Right...

KoBeWi commented:
The approach for appending errors needs to be reworked, as it's possible to stack errors from multiple nodes:
![image](https://github.com/godotengine/godot/assets/2223172/f7af57f8-998a-4d54-99c4-a0cd2c072268)

Calinou reviewed (approved):
Tested locally (rebased on top of `master` cb73a6e9f9046229a16b6b298c9086f03548e186), it works.

However, I can only confirm the first case (empty node name) is resolved, as I can't test the second issue. [I never see the **Batch Rename** option in the context menu when selecting several nodes](https://github.com/godotengine/godot/issues/78291#issuecomment-1603343062) if both are scene-unique, and this also applies in `master`.

Approving nonetheless, as this is an improvement over the existing situation.

KoBeWi commented:
> I never see the Batch Rename option in the context menu when selecting several nodes

Batch Rename does not appear if you select a parent and child node together. Likely a bug.

ajreckof commented:
> > I never see the Batch Rename option in the context menu when selecting several nodes
> 
> Batch Rename does not appear if you select a parent and child node together. Likely a bug.

See #76069

YuriSizov commented:
Besides a review from @KoBeWi this also needs a rebase.

ajreckof committed:
Fix batch rename for unique name and empty name.

ajreckof commented:
So I rebased this. The conflicting fix was about making sure it wouldn't raise an error when renaming a node to the same name when it is an unique name in owner. I moved this fix to the new places from this PR

YuriSizov commented:
I think the added meta should be prefixed with "_" by convention. Otherwise, this seems to address @KoBeWi's concern, albeit in a rather awkward way. Can't we set the meta and then compose a line by checking for each possible key, instead of doing this nested `if` thing?

KoBeWi commented on editor/gui/scene_tree_editor.cpp:
```diff
@@ -964,26 +964,55 @@ void SceneTreeEditor::_rename_node(Node *p_node, const String &p_name) {
+		return;
+	}
 
+	// we previously made sure name is not the same as current name so that it wonn't complain about already used unique name when not changing name
```
```suggestion
	// We previously made sure name is not the same as current name so that it won't complain about already used unique name when not changing name.
```

akien-mga replied:
Poke @ajreckof, would be good to fix this before merging.

KoBeWi commented:
> I think the added meta should be prefixed with "_" by convention.

This convention is irrelevant for internal editor nodes.

akien-mga commented:
Thanks!
